tmail-backend/jmap/extensions-rabbitmq/src/main/java/com/linagora/tmail/james/jmap/RabbitMQEmailAddressContactSubscriber.java

```diff
@@ -0,0 +1,98 @@
+
+package com.linagora.tmail.james.jmap;
+
+import static org.apache.james.backends.rabbitmq.Constants.AUTO_DELETE;
+import static org.apache.james.backends.rabbitmq.Constants.DURABLE;
+import static org.apache.james.util.ReactorUtils.publishIfPresent;
+
+import java.io.Closeable;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.apache.james.backends.rabbitmq.ReceiverProvider;
+import org.apache.james.events.EventBus;
+import org.apache.james.events.RegistrationKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableSet;
+import com.linagora.tmail.james.jmap.contact.EmailAddressContactMessage;
+import com.linagora.tmail.james.jmap.contact.TmailEvent;
+import com.linagora.tmail.james.jmap.json.EmailAddressContactMessageSerializer;
+import com.rabbitmq.client.Delivery;
+
+import reactor.core.Disposable;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Sinks;
+import reactor.core.scheduler.Schedulers;
+import reactor.rabbitmq.BindingSpecification;
+import reactor.rabbitmq.ExchangeSpecification;
+import reactor.rabbitmq.QueueSpecification;
+import reactor.rabbitmq.Receiver;
+import reactor.rabbitmq.Sender;
+
+public class RabbitMQEmailAddressContactSubscriber implements Closeable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RabbitMQEmailAddressContactSubscriber.class);
+
+    // TODO: Should parse from configure file
+    public static final String EXCHANGE_NAME = "EmailAddressContactExchange";
+    public static final String QUEUE_NAME = "EmailAddressContactQueue";
+    public static final String ROUTING_KEY = "EmailAddressContactRoutingKey";
+    private static final Set<RegistrationKey> EVENT_REGISTRATION_KEY = ImmutableSet.of();
+
+    private final ReceiverProvider receiverProvider;
+    private final Sinks.Many<EmailAddressContactMessage> listener;
+    private final Sender sender;
+    private final EventBus eventBus;
+    private Disposable listenQueueHandle;
+
+    @Inject
+    public RabbitMQEmailAddressContactSubscriber(ReceiverProvider receiverProvider, Sender sender,
+                                                 @Named(EmailAddressContactInjectKeys.AUTOCOMPLETE) EventBus eventBus) {
```

Comment:
We should enable providing connection to a distinct RabbitMQ (the contact app don't use James RabbitMQ but another one...)

Reply:
Why do we need the event bus? I thought we would index that straight away?

We are already taking this from a queue, it is already reliable...